apps/server/src/modules/fwu/uc/fwu.uc.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Logger } from '@src/core/logger';
+import { Configuration } from '@hpi-schul-cloud/commons';
+
+@Injectable()
+export class FwuUc {
+	constructor(private logger: Logger) {
+		this.logger.setContext(FwuUc.name);
+	}
+
+	async get(path: string): Promise<Uint8Array> {
+		const client = new S3Client({
+			endpoint: Configuration.get('FWU_CONTENT__S3_ENDPOINT') as string,
+			credentials: {
+				accessKeyId: Configuration.get('FWU_CONTENT__S3_ACCESS_KEY') as string,
+				secretAccessKey: Configuration.get('FWU_CONTENT__S3_SECRET_KEY') as string,
+			},
+			region: Configuration.get('FWU_CONTENT__S3_REGION') as string,
+			tls: true,
+			forcePathStyle: true,
+		});
+		const request = new GetObjectCommand({
+			Bucket: Configuration.get('FWU_CONTENT__S3_BUCKET') as string,
+			Key: path.toString(),
+		});
+		const response = await client.send(request);
+		if (response.$metadata.httpStatusCode !== 200) {
+			this.logger.warn({
+				message: 'S3 request failed for FWU content',
+				url: path,
+				error: response.$metadata.httpStatusCode,
+			});
+			return Promise.reject(response.$metadata.httpStatusCode);
+		}
+		const readableStream = response.Body as NodeJS.ReadableStream;
+		return new Promise<Uint8Array>((resolve, reject) => {
+			const chunks = [new Uint8Array()];
+			readableStream.on('data', (chunk: Uint8Array) => chunks.push(chunk));
+			readableStream.on('error', reject);
+			readableStream.on('end', () => resolve(Buffer.concat(chunks)));
+		});
```

Comment:
Please implenet file striming with 
https://docs.nestjs.com/techniques/streaming-files

For Example:

https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/files-storage/controller/files-storage.controller.ts#L99-L136

Reply:
NodeJS ReadableStream is replaced by Readable from the module 'stream', which is also used in the 's3-client.adapter.ts'.